<launch>
  <arg name="manager_name" default="transform_point_cloud_manager"/>
  
  <!-- manager nodelet  -->
  <node pkg="nodelet" type="nodelet" name="$(arg manager_name)"
	args="manager" clear_params="true" output="screen"/>

  <!-- point cloud transformation nodelet  -->
  <node pkg="nodelet" type="nodelet" name="transform_point_cloud"
	args="load grasp_multicam/TransformPointCloud $(arg manager_name)"
	clear_params="true" output="screen">
    <param name="fixed_frame" value="vision"/>
    <param name="transform_time_delay" value="-0.05"/>
    <remap from="~points" to="/monstar/points"/>
    <remap from="~transformed_points"  to="/monstar/transformed_points"/>
  </node>
  <!-- filter by altitude -->
  <node pkg="nodelet" type="nodelet" name="filter_point_cloud"
	args="load pcl/PassThrough $(arg manager_name)"
	clear_params="true" output="screen">
    <param name="filter_field_name" value="z"/>
    <!--
    <param name="filter_limit_min" value="-1.0"/>
    <param name="filter_limit_max" value="1.5"/>
    -->
    <param name="filter_limit_min" value="-10.0"/>
    <param name="filter_limit_max" value="10.5"/>
    <remap from="~input"  to="/monstar/transformed_points"/>
  </node>
</launch>
